p2p/host/basic/basic_host.go

```diff
@@ -81,8 +86,17 @@ func (h *BasicHost) newConnHandler(c inet.Conn) {
 }
 
 // newStreamHandler is the remote-opened stream handler for inet.Network
+// TODO: this feels a bit wonky
 func (h *BasicHost) newStreamHandler(s inet.Stream) {
-	h.Mux().Handle(s)
+	protoID, handle, err := h.Mux().ReadHeader(s)
+	if err != nil {
+		log.Error("protocol mux failed: %s", err)
+		return
+	}
```

Comment:
The problem was that calling `Handle` doesnt get me the protocol ID, and sinking all this into the Muxer gets ugly quick (i tried)